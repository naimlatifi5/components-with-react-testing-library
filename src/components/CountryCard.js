import React from 'react'

const CountryCard = ({ country }) => {
  return (
    <div className="country-cars">
      <div>Country name: {country.name}</div>
      <div>Country capital: {country.capital}</div>
      <div>Country population: {country.population}</div>
      <div>Country currency: {country.currency}</div>
    </div>
  )
}

export default CountryCard
