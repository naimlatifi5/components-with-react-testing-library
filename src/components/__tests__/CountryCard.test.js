import { render, screen } from '@testing-library/react'
import CountryCard from '../CountryCard'

describe('Country card component with snapshot testing', () => {
  test('It should match the snapshot', () => {
    const countries = {
      name: 'Sweden',
      capital: 'Stockholm',
      population: '10 million',
      currency: 'SEK',
    }
    const { asFragment } = render(<CountryCard country={countries} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
