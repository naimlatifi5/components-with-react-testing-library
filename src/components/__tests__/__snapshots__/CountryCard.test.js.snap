// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country card component with snapshot testing It should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="country-cars"
  >
    <div>
      Country name: Sweden
    </div>
    <div>
      Country capital: Stockholm
    </div>
    <div>
      Country population: 10 million
    </div>
    <div>
      Country currency: SEK
    </div>
  </div>
</DocumentFragment>
`;
